src/serializer/ResidualHeapSerializer.js

```diff
@@ -1692,13 +1692,26 @@ export class ResidualHeapSerializer {
     let remainingProperties = new Map(val.properties);
     const dummyProperties = new Set();
     let props = [];
+    let isFunction = f => f instanceof ECMAScriptSourceFunctionValue;
+    let isCertainlyLeaked = !val.mightNotBeHavocedObject();
+
+    // TODO #2259: Make deduplication in the face of leaking work for custom accessors
+    let shouldDropAsAssignedProp = (descriptor: Descriptor) =>
+      isCertainlyLeaked && !(isFunction(descriptor.get) || isFunction(descriptor.set));
```

Comment:
There is already a Spec method: IsDataDescriptor. It seems to me that isCertainlyLeaked && IsDataDescriptor(descriptor) would be safer and more in keeping with existing conventions.